### PR TITLE
Rebase Blood Drinker System, Arachne, Oneirophage

### PR DESCRIPTION
**Pull request by VMSolidus**
# Description
This is a simple rebase of the Blood Drinker System, and its related features that have been commented out and/or omitted due to its lack of rebase. I am NOT substantially updating any of this code at this time, outside of the barest minimum updates needed to make it run in the first place. The reason I am doing this is that I require the Blood Drinker system functional as a prerequisite for future features, and I will update or refactor it when needed. 

Arachne are still pending a Full Rework, but that is beyond the scope of this PR. 

# TODO

- [x] Make the code functional
- [x] Port Arachne
- [x] Uncomment Oneirophages
- [x] Re-add Oneirophage midround event

# Changelog

:cl:
- add: Arachne have been reimplemented!
- add: Oneirophages are back!

### CODE REVIEW

**Comment by VMSolidus** [comment]
DONE

**Comment by FoxxoTrystan** [comment]
What about the "normal" spooders?

File: Resources/Prototypes/Entities/Mobs/Species/arachne.yml
Comment (FoxxoTrystan) [inline Resources/Prototypes/Entities/Mobs/Species/arachne.yml:1]: Now That "Languages" got merged, specie should have a LanguageSpeaker Component.

Or they will go to the fallback as MobBase (if its is the right one)

Comment (VMSolidus) [reply Resources/Prototypes/Entities/Mobs/Species/arachne.yml:1]: They're using MobBaseHuman right now, so they are defaulted to Galactic Common + Sol Common. I haven't had a chance to really think about what language they should use to be fair.

File: Content.Server/Vampire/BloodSuckerComponent.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        [DataField, ViewVariables(VVAccess.ReadWrite)]
+        public TimeSpan Delay = TimeSpan.FromSeconds(4);
+
+        // ***INJECT WHEN SUCC***
+
+        /// <summary>
+        /// Whether to inject chems into a chemstream when we suck something.
+        /// </summary>
+        [DataField("injectWhenSucc")]
+        public bool InjectWhenSucc = false;
```
Comment (DangerRevolution) [inline Content.Server/Vampire/BloodSuckerComponent.cs:24]: Woah; cool possibities here for snake people

File: Content.Server/Vampire/Injector/BloodSuckerGlandInjectorComponent.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+namespace Content.Server.Vampiric
+{
+    [RegisterComponent]
+    /// <summary>
+    /// Item that gives a bloodsucker injection glands (for poison, usually)
```
Comment (DangerRevolution) [inline Content.Server/Vampire/Injector/BloodSuckerGlandInjectorComponent.cs:5]: Very cool

**Review by DangerRevolution — CHANGES_REQUESTED** [review]
They need a language to come with; and I also wanted to ask what the plan would be for Arachnids, what direction to take them if Arachnes become the vampires.

**Comment by VMSolidus** [comment]
> They need a language to come with; and I also wanted to ask what the plan would be for Arachnids, what direction to take them if Arachnes become the vampires.

So longer term, I'm actually going to comprehensively rework the Vampirism mechanic to later go with Arachne. I just needed this system rebased for other purposes, and changed as little as feasibly possible to get it working. 

Arachnids are meant to be parallel to it, it sort of fits in with a longer term goal that server hosts should just be able to toggle on or off any species they want for their server. If that means toggling off Arachnids and turning on Arachne, or vice versa, or having both enabled at the same time, that's on them. We can have defaults set(and the default for EE species is just, "All of them on by default".) 

For now I think I'm going to leave it as Sol Common in keeping with the "Humans & Human Mutants" theme. When I rework them it'll be fine for me to write a language just for Arachne.

**Review by DangerRevolution — CHANGES_REQUESTED** [review]
I'm actually going to say that Arachne *do* need a language; even if it's the lowest effort thing you can drum up where they just make different variants of "Hisssss"; people are going to want to play Arachne even if you don't see them as finished (AKA it's going to be merged pre-emptively by people) and a language would flesh them out further; while also expanding on the language system. Makes us look cooler.

**Comment by DangerRevolution** [comment]
> I'm actually going to say that Arachne _do_ need a language; even if it's the lowest effort thing you can drum up where they just make different variants of "Hisssss"; people are going to want to play Arachne even if you don't see them as finished (AKA it's going to be merged pre-emptively by people) and a language would flesh them out further; while also expanding on the language system. Makes us look cooler.

If you don't want to, I'll drum something up tomorrow morning.

**Comment by VMSolidus** [comment]
> > I'm actually going to say that Arachne _do_ need a language; even if it's the lowest effort thing you can drum up where they just make different variants of "Hisssss"; people are going to want to play Arachne even if you don't see them as finished (AKA it's going to be merged pre-emptively by people) and a language would flesh them out further; while also expanding on the language system. Makes us look cooler.
> 
> If you don't want to, I'll drum something up tomorrow morning.

Alright! I'm not opposed to you doing that, I'd actually really appreciate it if you did.

**Comment by DangerRevolution** [comment]
Completely forgot to make a language lol, now saw this as it's mirrored to Delta. Will make one in 12 hours...

**Comment by FoxxoTrystan** [comment]
> Completely forgot to make a language lol, now saw this as it's mirrored to Delta. Will make one in 12 hours...

If mutan they should use Sol Common but also if you do have a good design for a spooder langauge il give it to others spiders, tho if no design idea come in mind , i have one already.

**Comment by DangerRevolution** [comment]
> > Completely forgot to make a language lol, now saw this as it's mirrored to Delta. Will make one in 12 hours...
> 
> If mutan they should use Sol Common but also if you do have a good design for a spooder langauge il give it to others spiders, tho if no design idea come in mind , i have one already.

honestly idm; languages can always be re-visited with lore, whatever works

**Review by DangerRevolution — APPROVED** [review]

**Review by OldDanceJacket — APPROVED** [review]

**Pull request merged**